net-tools/netwatch/src/udp.rs

```diff
@@ -89,9 +712,11 @@ impl UdpSocket {
 
         // Convert into tokio UdpSocket
         let socket = tokio::net::UdpSocket::from_std(socket).context("conversion to tokio")?;
+        let socket_ref = quinn_udp::UdpSockRef::from(&socket);
+        let socket_state = quinn_udp::UdpSocketState::new(socket_ref)?;
 
+        let local_addr = socket.local_addr().context("local addr")?;
         if addr.port() != 0 {
-            let local_addr = socket.local_addr().context("local addr")?;
             ensure!(
```

Comment:
Eh - I think this was me who moved the `local_addr` call accidentally (I used it when tracing stuff, but removed only the tracing call).

I think it can move back into the `if`.

Reply:
no I moved this on purpose, we store the `local_addr` here now

Reply:
I swear I can read code. :woozy_face: